WordPress/openverse
# Add basic API load testing script

## Description


This adds a basic locust load testing script for the API. It's pretty naive but the idea is to just hit the API very hard and force it to fall over. So far I haven't been able to make it actually fall over, maybe the user count needs to go up into the hundreds. I also haven't tested against v2.5.3 which was the problematic version.

## Testing Instructions

Checkout this branch and `cd load_testing`. Run `just install` and then `just api` (make sure you have the API running locally). Confirm via `docker stats` that the API is being hit with a bunch of requests. `just api` won't exit on it's own so use "control + c" to stop it or pass a time argument like `just api -t 10m` (this will run the load test for 10 minutes).

## Checklist

- [N/A] I added or updated tests for the changes I made (if applicable).

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
